### PR TITLE
This commit deals with cleaning up of the Common code base. It does the

### PR DESCRIPTION
**Pull request by rishirsinha**
following till now:

1. Removes usage of Common.h from inside the Common folder.
2. It has removed the dependencies for Global unique variables from
FabricCommon. This came in from APs change in the WindowsFabric repo.
3. It has move Tracing and JsonSerialization to its own folders.

### CODE REVIEW

**Comment by MansoorMohsin** [comment]
@rishirsinha

File: build/Dockerfile.ubuntu [hunk 1]
```diff
@@ -36,6 +36,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     liblttng-ctl-dev \
     libncurses5-dev \
     libncursesw5-dev \
+    libpython3-dev \
```
Comment (MansoorMohsin) [inline build/Dockerfile.ubuntu]: I've been using `python3-dev` instead and it works fine for me.

File: src/prod/CMakeLists.txt [hunk 2]
```diff
@@ -23,6 +23,7 @@ set(EXECUTABLE_OUTPUT_DIR ${PROJECT_BINARY_DIR}/exe CACHE STRING "Executable out
 
 # Horizon static library names
 set(lib_Common "Common" CACHE STRING "Common library")
+set(lib_Tracing "Tracing" CACHE STRING "Tracing library")
```
Comment (MansoorMohsin) [inline src/prod/CMakeLists.txt]: Consider using `Common.Tracing` instead. That's the convention we've been following under `Reliability`.

File: src/prod/ktl/src/CMakeLists.txt [hunk 3]
```diff
@@ -246,7 +246,4 @@ set (lib_KxmUser "KxmUser" CACHE STRING "KXM User API Library")
 #    )
 
 add_subdirectory (src)
-#add_subdirectory (tools)
-add_subdirectory (logger)
-#add_subdirectory (test)
-add_subdirectory (linuxsetup)
+#add_subdirectory (logger)
```
Comment (MansoorMohsin) [inline src/prod/ktl/src/CMakeLists.txt]: Why not just delete this? We can always later if needed.

File: src/prod/shared/ktlwf/com/inc/AsyncCallInAdapter.h [hunk 4]
```diff
@@ -8,6 +8,9 @@
 
 #include "FabricAsyncContextBase.h"
 #include "KComUtility.h"
+#include "Common/ComUnknownBase.h"
```
Comment (MansoorMohsin) [inline src/prod/shared/ktlwf/com/inc/AsyncCallInAdapter.h]: So far, we've been avoiding including header files in header files (unless it's a PCH). I'm not sure what the right pattern should be here.

File: src/prod/shared/ktlwf/com/src/AsyncCallInAdapter.cpp [hunk 5]
```diff
@@ -9,6 +9,7 @@
 #include "Common/Common.h"
```
Comment (MansoorMohsin) [inline src/prod/shared/ktlwf/com/src/AsyncCallInAdapter.cpp]: How much of `Common` do we need here? Can we just add what's needed?

File: src/prod/src/CMakeLists.txt [hunk 6]
```diff
@@ -1,4 +1,5 @@
 add_subdirectory (Common)
+add_subdirectory (Common/Tracing)
```
Comment (MansoorMohsin) [inline src/prod/src/CMakeLists.txt]: Instead, why not have `Common` add the `Tracing` directory?

File: src/prod/shared/ktlwf/com/src/AsyncCallInAdapter.cpp (hunk 5 above)
Comment (rishirsinha) [reply src/prod/shared/ktlwf/com/src/AsyncCallInAdapter.cpp]: I have not made any major fixes to ktl. That is next round.

File: src/prod/shared/ktlwf/com/inc/AsyncCallInAdapter.h (hunk 4 above)
Comment (rishirsinha) [reply src/prod/shared/ktlwf/com/inc/AsyncCallInAdapter.h]: The model I am trying to get to is the one suggested by IWYU, which is keep the header file as localized as possible. I am hoping after the splitting of Common, I can make such headers for the smaller libs and add pch for them.

File: build/Dockerfile.ubuntu (hunk 1 above)
Comment (rishirsinha) [reply build/Dockerfile.ubuntu]: With Alex's latest changes I believe libppython will also be needed.

File: src/prod/CMakeLists.txt (hunk 2 above)
Comment (rishirsinha) [reply src/prod/CMakeLists.txt]: Next Commit

File: src/prod/src/CMakeLists.txt (hunk 6 above)
Comment (rishirsinha) [reply src/prod/src/CMakeLists.txt]: Next commit.

File: src/prod/ktl/src/CMakeLists.txt (hunk 3 above)
Comment (rishirsinha) [reply src/prod/ktl/src/CMakeLists.txt]: Done.

**Pull request merged**